### PR TITLE
Fix more TODOs. CompactMerkleTree returns an error if loading fails

### PR DESCRIPTION
**Pull request by Martin2112**
Plus tests. I want to add tests that loading fetches the right nodes but I'll do that in a later PR.

### CODE REVIEW

File: merkle/compact_merkle_tree.go [hunk 1]
```diff
@@ -4,11 +4,21 @@ import (
...
 	"github.com/google/trillian"
 )
 
+type RootHashMismatchError struct {
+	ExpectedHash []byte
+	ActualHash   []byte
+}
+
+func (r RootHashMismatchError) Error() string {
+	return fmt.Sprintf("Root hash mismatch got: %v expected: %v", r.ExpectedHash, r.ActualHash)
```
Comment (AlCutter) [inline merkle/compact_merkle_tree.go]: initial lowercase: "root hash mis..."

File: merkle/compact_merkle_tree_test.go [hunk 2]
```diff
@@ -110,3 +112,32 @@ func TestAddingLeaves(t *testing.T) {
...
+// matter what it is but it must be correct length for an SHA256 hash as if it was real
+func fixedHashGetNodeFunc(depth int, index int64) (trillian.Hash, error) {
+	return []byte("12345678901234567890123456789012"), nil
+}
+
+func TestLoadingTreeFailsNodeFetch(t *testing.T) {
+	_, err := NewCompactMerkleTreeWithState(trillian.NewSHA256(), 237, failingGetNodeFunc, []byte("notimportant"))
+
+	if err == nil || !strings.Contains(err.Error(), "Bang!") {
+		t.Fatalf("Did not return correctly on failed node fetch")
```
Comment (AlCutter) [inline merkle/compact_merkle_tree_test.go]: log the error/nil to help with debugging?

File: merkle/compact_merkle_tree_test.go [hunk 3]
```diff
@@ -110,3 +112,32 @@ func TestAddingLeaves(t *testing.T) {
...
+}
+
+func TestLoadingTreeFailsBadRootHash(t *testing.T) {
+	// Supply a root hash that can't possibly match the result of the SHA 256 hashing on our dummy
+	// data
+	_, err := NewCompactMerkleTreeWithState(trillian.NewSHA256(), 237, fixedHashGetNodeFunc, []byte("nomatch!nomatch!nomatch!nomatch!"))
+	_, ok := err.(RootHashMismatchError)
+
+	if err == nil || !ok {
+		t.Fatalf("Did not return correct error type on root mismatch")
```
Comment (AlCutter) [inline merkle/compact_merkle_tree_test.go]: same here?

File: merkle/compact_merkle_tree.go (hunk 1 above)
Comment (Martin2112) [inline merkle/compact_merkle_tree.go]: Done.

File: merkle/compact_merkle_tree_test.go (hunk 2 above)
Comment (Martin2112) [inline merkle/compact_merkle_tree_test.go]: Done.

File: merkle/compact_merkle_tree_test.go (hunk 3 above)
Comment (Martin2112) [inline merkle/compact_merkle_tree_test.go]: Done.

**Pull request merged**